### PR TITLE
Test Coverage: Improve test coverage for syft.core.io.address

### PR DESCRIPTION
**Pull request by nahuakang**
## Description
This PR adds more unit tests for `src/syft/core/io/address.py` and is related to [Issue 4805](https://github.com/OpenMined/PySyft/issues/4805) for improving test coverage.

`address.py`'s coverage improved from 64% to 96%:
```
src/syft/core/io/address.py                         164      7    96%   25, 100, 104, 108, 118, 351-352
```

`src/syft/core/io`'s coverage improved from 70% to 82.27%:
```
----------- coverage: platform linux, python 3.8.6-final-0 -----------
Name                                              Stmts   Miss  Cover   Missing
-------------------------------------------------------------------------------
src/syft/core/io/__init__.py                          0      0   100%
src/syft/core/io/address.py                         164      7    96%   25, 100, 104, 108, 118, 351-352
src/syft/core/io/connection.py                       71     23    68%   16, 21, 26, 32, 38, 44, 48, 52, 56, 61, 67, 72, 77, 81, 85, 89, 94, 100, 106, 112, 116, 120, 124
src/syft/core/io/location/__init__.py                 6      0   100%
src/syft/core/io/location/group/__init__.py           0      0   100%
src/syft/core/io/location/group/group.py              5      1    80%   19
src/syft/core/io/location/group/registry.py           7      2    71%   21-22
src/syft/core/io/location/group/subscription.py       7      2    71%   11-12
src/syft/core/io/location/location.py                28      4    86%   27, 37, 55, 70
src/syft/core/io/location/specific.py                28      0   100%
src/syft/core/io/route.py                            61     21    66%   119, 124-128, 132, 136, 141, 146, 151, 160-161, 166-167, 172, 177, 181, 188-190, 206
src/syft/core/io/virtual.py                          63     18    71%   32, 38, 44, 50, 54, 58-59, 65, 72, 77, 83, 89, 93, 97, 103, 109-112
-------------------------------------------------------------------------------
TOTAL                                               440     78    82%
```

## Affected Dependencies
N/A

## How has this been tested?
`pytest tests/syft/core/io --cov-report term --cov=src/syft/core/io -m fast -n auto`

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: tests/syft/core/io/address_test.py [hunk 1]
```diff
@@ -159,6 +140,242 @@ def test_to_string() -> None:
...
+    domain = SpecificLocation(id=an_id)
+    device = SpecificLocation(id=an_id)
+    vm = SpecificLocation(id=an_id)
+    address = Address(
+        network=network,
+        domain=domain,
+        device=device,
+        vm=vm,
+    )
+    assert address.addressables == [vm, device, domain, network]
```
Comment (tudorcebere) [inline tests/syft/core/io/address_test.py:370]: Great job overall! I think this should be enough as a test, setting the others to None might not be testing something. I would leave only line 370.

**Review by tudorcebere — COMMENTED** [review]
Great job! One tiny comment and it's good to go! Thanks for addressing this @nahuakang, amazing work.

**Comment by madhavajay** [comment]
This is excellent work @nahuakang the code looks nice and clean, the coverage is improved and it passes all the lint checks! ❤️

File: tests/syft/core/io/address_test.py (hunk 1 above)
Comment (madhavajay) [reply tests/syft/core/io/address_test.py:370]: The coverage doesn't change but it is the only place I can see where we are testing that the internal state of `address.addressables` is affected by those properties. In fact we don't use that property anywhere, so maybe we should just delete it? 🤷🏼‍♀️ I would say delete or keep the additional tests, either way.

Comment (nahuakang) [reply tests/syft/core/io/address_test.py:370]: Addressed. Thanks!

**Comment by nahuakang** [comment]
I'll keep working in the `core/io` directory and make the overall coverage to be around 90% at least :smile:

**Review by madhavajay — APPROVED** [review]
LGTM! ❤️

**Pull request merged**